File: src/dataset/python/matrix_exercism-ThomasZumsteg/claude/ClaudeSonnet4_matrix_v1.py
"""A 2 dimentional matrix object"""

class Matrix(object):
    """Creates a 2d matrix object with accessable rows and columns"""

    def __init__(self, matrix):
        """Copy of the input data"""
        self.matrix = matrix
        self._rows = None
        self._columns = None

    @property
    def rows(self):
        """Matrix displayed as rows, coumns"""
        if self._rows is None:
            self._rows = [[int(n) for n in row.split()]
                          for row in self.matrix.split("\n")]
        return self._rows

    @property
    def columns(self):
        """Matrix displayed as columns, rows"""
        if self._columns is None:
            rows = self.rows
            num_cols = len(rows[0])
            self._columns = [[] for _ in range(num_cols)]
            for row in rows:
                for i, item in enumerate(row):
                    self._columns[i].append(item)
        return self._columns

File: src/dataset/javascript/matrix_exercism-ThomasZumsteg/claude/ClaudeSonnet4_matrix_v1.js
var Matrix = function(matrix) {
	/* A 2d matrix class */
	const lines = matrix.split("\n");
	const numRows = lines.length;
	const numCols = lines[0] ? lines[0].split(" ").length : 0;
	
	this.rows = new Array(numRows);
	for (let i = 0; i < numRows; i++) {
		const rowElements = lines[i].split(" ");
		this.rows[i] = new Array(numCols);
		for (let j = 0; j < numCols; j++) {
			this.rows[i][j] = parseInt(rowElements[j], 10);
		}
	}

	this.columns = new Array(numCols);
	for (let j = 0; j < numCols; j++) {
		this.columns[j] = new Array(numRows);
		for (let i = 0; i < numRows; i++) {
			this.columns[j][i] = this.rows[i][j];
		}
	}
}

module.exports = Matrix;